shared/libraries/opcua/opcuaserver/src/opcuaaddnodeparams.cpp

```diff
@@ -2,6 +2,19 @@
 
 BEGIN_NAMESPACE_OPENDAQ_OPCUA
 
+/*RequestedNodeIdBaseOnName*/
+
+static daq::opcua::OpcUaNodeId RequestedNodeIdBaseOnName(const std::string& name, const OpcUaNodeId& parentNodeId)
+{
+    if (parentNodeId.getValue().identifierType == UA_NODEIDTYPE_STRING)
+    {
+        std::string newNodeIdStr = utils::ToStdString(parentNodeId.getValue().identifier.string) + "/" + name;
+        return OpcUaNodeId(parentNodeId.getNamespaceIndex(), newNodeIdStr);
```

Comment:
I'm not entirely sure that the child node ID is always the same as the parent's here.

Reply:
I thought through it again, and I am 99,9% sure ;-). 
Because: the path is always built up as: <parent_id>/<childName>
Since openDAQ object have already every time a string nodeId, like IO, etc. we can use this string path + the child's name. 
If you have nested properties, also a property has a string id first, because it belongs to one of our objects. 

The only difference I see, based on your feedback from the other comment that a nodeID would look like this than:

Component1/Component2/Prop1/Prop2

and not like this:

Component1/Component2.Prop1.Prop2 as it is in the core.

Reply:
Ah, I might've worded my comment a bit oddly. I meant the ID due to the namespace index. We should probably use the namespace 0 index here?

Reply:
0 index is normaly reserved for the OPC-UA standard. See:

Namespace 0 is defined by the OPC Foundation and contains OPC Core nodes that you cannot modify.

Namespace 1 is specific to your server can contains server info like certificates and open connections.

Namespace 2 ... n is where your business objects are.

And all our objects are in Namespace 2 and greater so it is right...